test_scripts/ATF_ShutDown_IGNITION_OFF.lua

```diff
@@ -0,0 +1,53 @@
+Test = require('user_modules/dummy_connecttest')
+require('cardinalities')
+local mobile_session = require('mobile_session')
+local module = require('testbase')
+local commonFunctions = require('user_modules/shared_testcases/commonFunctions')
+local commonSteps = require('user_modules/shared_testcases/commonSteps')
+require('user_modules/AppTypes')
+
+--- Preconditions
+commonFunctions:newTestCasesGroup("Preconditions")
+
+commonSteps:DeleteLogsFileAndPolicyTable()
+
+function Test:StartSDL_with_one_activated_App()
+  self:runSDL()
+  commonFunctions:waitForSDLStart(self):Do(function()
+    self:initHMI():Do(function()
+      print("Hmi initialized")
+      self:initHMI_onReady():Do(function ()
+        print("Hmi is ready")
+        self:connectMobile():Do(function ()
+          print("Mobile Connected")
+          self:startSession():Do(function ()
+            commonSteps:ActivateAppInSpecificLevel(self, self.applications["Test Application"])
+            EXPECT_NOTIFICATION("OnHMIStatus", {hmiLevel = "FULL", systemContext = "MAIN"})
+          end)
+        end)
+      end)
+    end)
+  end)
+end
+
+--- Tests
+--- Description:
+---  HMI sends OnExitAllApplications with reason IGNITION_OFF:
+---  1. SDL sends to HMI OnSDLClose
+---  2. SDL sends to app OnAppInterfaceUnregistered
+
+commonFunctions:newTestCasesGroup("Check that SDL finish it's working properly by IGNITION_OFF")
+function Test:ShutDown_IGNITION_OFF()
+  self.hmiConnection:SendNotification("BasicCommunication.OnExitAllApplications",
+    { reason = "IGNITION_OFF" })
+  EXPECT_NOTIFICATION("OnAppInterfaceUnregistered", { reason = "IGNITION_OFF" })
+  EXPECT_HMINOTIFICATION("BasicCommunication.OnAppUnregistered", { unexpectedDisconnect = false }) 
+  --ATF is unable to receive SDL notificaion
+  --EXPECT_HMINOTIFICATION("BasicCommunication.OnSDLClose")
+end
+
+--- Postcondition
+commonFunctions:newTestCasesGroup("Postcondition")
+function Test:Stop_SDL()
```

Comment:
@LuxoftAKutsan, I agree that in template we are using :.
But if you run LuaCheck you will have error and if you try to create commit using Prehook Commit, creation of commit will not be allowed.
For now Prehook commit was used only for Policy scripts, but I think that in future it will be applicable for all scripts.